src/uart.c

```diff
@@ -3,6 +3,24 @@
 
 #include <metal/uart.h>
 
+struct metal_uart *metal_uart_get_device(void)
+{
+#ifdef __METAL_DT_STDOUT_UART_HANDLE
+    return __METAL_DT_STDOUT_UART_HANDLE;
+#else
```

Comment:
The stdout handle should only matter to the TTY api, what we want here is to have a table of uart devices that we can access, just like you can see with the SPI, etc. The way to do that is to do a corresponding PR to freedom-devicetree-tools which adds the table.

Reply:
Yes. totally agree. I missed the TTY stuff. TTY APIs could meet my current need. I should re-sent the PR only for PMP stuff. Thanks for your advise.